File: java/test/src/main/java/test/Ice/proxy/MyDerivedClassI.java
// Copyright (c) ZeroC, Inc.

package test.Ice.proxy;

import com.zeroc.Ice.Current;
import com.zeroc.Ice.ObjectPrx;

import test.Ice.proxy.Test.MyDerivedClass;

import java.util.Map;

public final class MyDerivedClassI implements MyDerivedClass {
    public MyDerivedClassI() {
    }

    @Override
    public ObjectPrx echo(ObjectPrx obj, Current c) {
        return obj;
    }

    @Override
    public void shutdown(Current c) {
        c.adapter.getCommunicator().shutdown();
    }

    @Override
    public Map<String, String> getContext(Current current) {
        return _ctx;
    }

    @Override
    public boolean ice_isA(String s, Current current) {
        _ctx = current.ctx;
        return MyDerivedClass.super.ice_isA(s, current);
    }

    private Map<String, String> _ctx;
}

File: csharp/test/Ice/proxy/MyDerivedClassI.cs
// Copyright (c) ZeroC, Inc.

namespace Ice.proxy
{


        public sealed class MyDerivedClassI : Test.MyDerivedClassDisp_, Ice.Object
        {
            public MyDerivedClassI()
            {
            }

            public override Ice.ObjectPrx echo(Ice.ObjectPrx obj, Ice.Current c)
            {
                return obj;
            }

            public override void shutdown(Ice.Current current)
            {
                current.adapter.getCommunicator().shutdown();
            }

            public override Dictionary<string, string> getContext(Ice.Current current)
            {
                return _ctx;
            }

            public bool ice_isA(string s, Ice.Current current)
            {
                _ctx = current.ctx;
                // TODO: call the default implementation when the base(Object) syntax is supported.
                return Array.BinarySearch((this as Ice.Object).ice_ids(current), s) >= 0;
            }

            private Dictionary<string, string> _ctx;
        }
    }

